### DISCUSSION TITLE
Help with WithAttribute?

### DISCUSSION THREAD

**Discussion by AlmostInteractive** [NoeticTools/Net2HassMqtt#1] [General]
Could you share an example of how to use `WithAttribute` and then how to retrieve that data in HA?

**Comment by RobSmyth** [comment]
Hi, yea attributes are not obvious in HASS.

The readme example is:

```csharp
    device.HasBinarySensor(config => config.OnModel(irrigationProgram)
                                                        :
                                           .WithDurationAttribute(nameof(prog.Duration), "Duration (Minutes)", DurationSensorUoM.Minutes)
                                           .WithAttribute(nameof(prog.ZoneIds), "Zone IDs"))
```

Another example to flesh it out a bit:

```
        device.HasBatteryChargingBinarySensor(config => config.OnModel(model)
                                                              .WithStatusProperty(nameof(QuickStartDemoModel.BatteryCharging))
                                                              .WithFriendlyName("Battery Charging Status")
                                                              .WithNodeId("battery_1_charging")
                                                              .WithAttribute("PROPERTY_NAME", "NAME", "UNIT_OF_MEASURE"));
```

Attributes are read from properties on the same model that holds the entity's property. They are only read when sending the entity value.

If you add attributes to an entity then the entities property dialog will show a drop down list of attributes as shown below. They can be used by scripts and some controls but I'm not used with them. (that I can remember).

![image](https://github.com/user-attachments/assets/2400380a-3f90-46cc-ac71-9399c7034798)


**When updating entities remember to refresh HASS page**

I will update the readme.

**Comment by AlmostInteractive** [comment]
I understand now!  Thank you very much for that explanation.
